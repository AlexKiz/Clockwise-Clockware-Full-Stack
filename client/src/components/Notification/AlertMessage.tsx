import React, {useState, FC, useEffect} from 'react';
import {Alert, IconButton, Snackbar} from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import {AlertProps} from './componentConstants';


const AlertMessage: FC<AlertProps> = (props) => {
	const [open, setOpen] = useState<boolean>(true);

	useEffect(() => {
		setOpen(props.notify);
	});

	return (
		<Snackbar
			anchorOrigin={{vertical: 'top', horizontal: 'right'}}
			open={open}
			autoHideDuration={6000}
			onClose={()=>{
				setOpen(false);
				props.isOpen(false);
			}}
		>
			<Alert
				severity={props.alertType}
				action={
					<IconButton
						aria-label="close"
						color="inherit"
						size="small"
						onClick={() => {
							setOpen(false);
							props.isOpen(false);
						}}
					>
						<CloseIcon fontSize="inherit" />
					</IconButton>
				}
			>
				{props.message}
			</Alert>
		</Snackbar>
	);
};

export default AlertMessage;
