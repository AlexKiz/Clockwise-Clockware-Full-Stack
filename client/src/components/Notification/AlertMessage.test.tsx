import React from 'react';
import AlertMessage from './AlertMessage';

import {render} from '@testing-library/react';

it('matches snapshot', () => {
	const {asFragment} = render(<AlertMessage />);// TODO: props

	expect(asFragment()).toMatchSnapshot();
});
