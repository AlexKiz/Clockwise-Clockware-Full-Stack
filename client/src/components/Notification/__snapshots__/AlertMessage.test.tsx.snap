// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginTopRight css-1arijsr-MuiSnackbar-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-efvcc7-MuiPaper-root-MuiAlert-root"
      direction="down"
      role="alert"
      style="opacity: 0; transform: scale(0.75, 0.5625); transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 130ms cubic-bezier(0.4, 0, 0.2, 1) 65ms;"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="SuccessOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-acap47-MuiAlert-message"
      />
      <div
        class="MuiAlert-action css-ki1hdl-MuiAlert-action"
      >
        <button
          aria-label="close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
